src/arith/rewrite_simplify.cc

```diff
@@ -820,6 +820,10 @@ PrimExpr RewriteSimplifier::Impl::VisitExpr_(const FloorDivNode* op) {
     TVM_TRY_REWRITE_IF(floordiv(x * y, y), x, CanProveGreaterEqual(y.Eval(), 0));
     TVM_TRY_REWRITE_IF(floordiv(y * x, y), x, CanProveGreaterEqual(y.Eval(), 0));
 
+    if ((floordiv(x, y)).Match(ret) && analyzer_->CanProve(x.Eval() < y.Eval())) {
+      return 0;
+    }
+
```

Comment:
I'm not sure what this is for?

Reply:
Maybe need to a a pass test for this.

Reply:
This is needed so that the simplifier here: https://github.com/apache/tvm/pull/9873/files#diff-04a1aab966320b6f63c390cc8b79f79b543a1a7a3f560ea3a97d18c2bb041a0aR837-R839 will remove the division by groups. Without this, simplification get stuck simplifying `ff // (num_filter // groups) * (in_channel // groups) + rc` to `ff // num_filter * in_channel + rc`. It can't prove that `ff // num_filter = 0`.